doc/PLUGIN.md

```diff
@@ -0,0 +1,44 @@
+# Plugin API
+
+`Miradord` makes use of external plugins to define and enforce spend
+policies.
```

Comment:
```suggestion
`miradord` makes use of external plugins to define its re-vaulting policy (decide whether to broadcast a Cancel transaction for a given Unvault).
```

1. The plugins don't enforce the policy, the watchtower does by broadcasting the Cancel
2. It's not really a spend policy, and it's hard to define. So i think a good way to describe is by going the opposite way: the "Cancel policy".

Reply:
I think it's worth expanding on the design before going on with the specifics of each call.

> The plugin process is kept super simple.
> The watchtower expects to be able to start a given plugin as a child process. It must therefore be executable (whether it is a binary or, say, a script with a shebang). The path to the executable must be set in the configuration file. Additional configuration specific to the plugin may be specified in `miradord`'s configuration file. It will be passed to the plugin upon every request.
> The watchtower will start the child process, write JSON to its `stdin`, read a JSON response from its `stdout` and then wait for the process to terminate.
> The watchtower may poll more than one plugin. In this case it will only let an Unvault go through if none of the plugins told it to broadcast the Cancel transaction.